### PR TITLE
Generate new types

### LINKED ISSUE #80: PangoAttribute support
Looks like [Text Attributes](https://developer.gnome.org/pango/stable/pango-Text-Attributes.html) doesn't implemented at all. Does it easily implemented?

Personally, I need only `pango_attr_underline_new`, `pango_attr_letter_spacing_new` and `pango_attr_strikethrough_new`.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #80.

Waiting for https://github.com/gtk-rs/gir/pull/397 to be merged.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Waiting for https://github.com/gtk-rs/gir/pull/398 now.

**Comment by GuillaumeGomez** [comment]
@EPashkin: this one should be ready now.

**Comment by EPashkin** [comment]
@GuillaumeGomez Sorry, I check this in hour.

**Comment by GuillaumeGomez** [comment]
No hurry, take your time (and thanks!).

**Comment by EPashkin** [comment]
`pango_font_description_from_string` has nonoptional return
`pango_font_description_set_family_static` etc. seems better ignored
Don't see way to create `AttrClass`

**Comment by GuillaumeGomez** [comment]
It's given by a function iirc. I'll check this evening.

**Comment by GuillaumeGomez** [comment]
Updated. I added the missing getter for `AttrClass` (you're not supposed to create it).

Also, what's wrong with `pango_font_description_set_family_static`, etc.?

File: src/font_description.rs [hunk 1]
```diff
@@ -12,3 +12,11 @@ impl FromGlibPtrNone<*const ffi::PangoFontDescription> for FontDescription {
         from_glib_none(mut_override(ptr))
     }
 }
+
+impl FontDescription {
+    pub fn from_string(str: &str) -> FontDescription {
+        unsafe {
+            from_glib_full(ffi::pango_font_description_from_string(str.to_glib_none().0))
```
Comment (EPashkin) [inline src/font_description.rs]: This can be done by gir: https://github.com/gtk-rs/gdk/blob/master/Gir.toml#L176-L182

Comment (GuillaumeGomez) [inline src/font_description.rs]: Oh nice!

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by EPashkin** [comment]
By https://developer.gnome.org/pango/stable/pango-Fonts.html#pango-font-description-set-family-static
`family` need be static string. Not checked other but possibility that others has same сonstraints

**Comment by GuillaumeGomez** [comment]
So I checked, it seems to be the only function which requires to have an argument with a static lifetime.

**Comment by EPashkin** [comment]
IMHO others is same: return value actual only when argument live.
https://developer.gnome.org/pango/stable/pango-Fonts.html#pango-font-description-copy-static
https://developer.gnome.org/pango/stable/pango-Fonts.html#pango-font-description-merge-static

**Comment by GuillaumeGomez** [comment]
Hum... I'll put a lifetime on the returned values as well.

File: src/font_description.rs [hunk 2]
```diff
@@ -1,158 +1,22 @@
...
         }
     }
-}
-
-impl Drop for FontDescription {
-    fn drop(&mut self) {
-        unsafe { ffi::pango_font_description_free(self.0) }
-    }
-}
+}
```
Comment (EPashkin) [inline src/font_description.rs]: Better add extra newline

**Comment by GuillaumeGomez** [comment]
It's sad, but I had to remove the "_static" functions... Too lazy to add a `PhantomData`.

**Comment by GuillaumeGomez** [comment]
I should maybe stop here instead of adding new types again and again...

File: src/context.rs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+use Context;
+use FontDescription;
+use FontMetrics;
+use Language;
+use ffi;
+use glib::translate::*;
+
+impl Context {
+    pub fn get_metrics<'a, 'b, P: Into<Option<&'a FontDescription>>, Q: Into<Option<&'b Language>>>(&self, desc: P, language: Q) -> Option<FontMetrics> {
```
Comment (GuillaumeGomez) [inline src/context.rs]: @EPashkin: in here too, `Q` should be `Into<Option<&mut 'b Language>>>`. :p

File: src/font.rs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+// DO NOT EDIT
+
+use Font;
+use FontMetrics;
+use Language;
+use ffi;
+use glib::translate::*;
+
+impl Font {
+    pub fn get_metrics<'a, P: Into<Option<&'a Language>>>(&self, language: P) -> Option<FontMetrics> {
```
Comment (GuillaumeGomez) [inline src/font.rs]: And here as well: `Into<Option<&mut 'b Language>>>`.

**Comment by EPashkin** [comment]
https://developer.gnome.org/pango/stable/pango-Text-Processing.html#pango-context-get-metrics
and
https://developer.gnome.org/pango/stable/pango-Fonts.html#pango-font-get-metrics
also can be handled by Gir with `const` on parameter.

**Comment by GuillaumeGomez** [comment]
It's magical! :D

**Comment by GuillaumeGomez** [comment]
And updated.

**Comment by EPashkin** [comment]
https://developer.gnome.org/pango/stable/pango-Text-Processing.html#pango-context-set-language
also don't need mut. And others with `language.to_glib_none_mut()` too.
As you continue adding new objects maybe add "Pango.Fontset" too?

**Comment by GuillaumeGomez** [comment]
I remember having an issue with FontSet but I'll try re-binding it this evening.

**Comment by GuillaumeGomez** [comment]
Updated!

**Comment by EPashkin** [comment]
Still `language.to_glib_none_mut()` in many places
and IMHO unneeded manual `get_metrics` for `Font` and `Context`

**Comment by GuillaumeGomez** [comment]
Oh damn, I overwrote the automatic generation...

**Comment by GuillaumeGomez** [comment]
No more `.to_glib_none_mut()` on `language` normally.

**Comment by EPashkin** [comment]
👍
Thanks @GuillaumeGomez

**Pull request merged**